solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -353,86 +323,55 @@ protected NamedList loadConfigSetFlags(CoreDescriptor cd, SolrResourceLoader loa
   public abstract void deleteConfig(String configName) throws IOException;
 
   /**
-   * Delete a file in config
+   * Delete files in config
    *
-   * @param configName the config to delete
-   * @param fileName the file to delete
-   * @throws IOException if an I/O error occurs
-   */
-  public abstract void deleteFileFromConfig(String configName, String fileName) throws IOException;
-
-  /**
-   * Set config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be set in config
-   * @throws IOException if an I/O error occurs
-   */
-  public abstract void setConfigMetadata(String configName, Map<Object, Object> data) throws IOException;
-
-  /**
-   * Set config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be set in config
+   * @param filesToDelete a list of file paths to delete
    * @throws IOException if an I/O error occurs
    */
-  public abstract void setConfigMetadata(String configName, byte[] data) throws IOException;
+  public abstract void deleteFilesFromConfig(List<String> filesToDelete) throws IOException;
 
   /**
-   * Update config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be updated in config
-   * @throws IOException if an I/O error occurs or if config does not exist
-   */
-  public abstract void updateConfigMetadata(String configName, Map<Object, Object> data) throws IOException;
-
-  /**
-   * Update config metadata
-   *
-   * @param configName the config
-   * @param data metadata to be updated in config
+   * Set the config metadata
+   * If config does not exist, it will be created and set metadata on it
```

Comment:
Wow; this creates a config.  That's unexpected I think, given its name.  But I can sorta see how this is and maybe all the methods on this class that upload data will, in effect, create a config by virtue of setting data.  Ok.